Library/Homebrew/cask/test/cask/artifact/alt_target_test.rb

```diff
@@ -1,85 +1,79 @@
 require "test_helper"
 
 describe Hbc::Artifact::App do
-  let(:local_alt_caffeine) {
-    Hbc.load("with-alt-target").tap do |cask|
+  describe "activate to alternate target" do
+    let(:cask) { Hbc.load("with-alt-target") }
+
+    let(:install_phase) {
+      lambda { Hbc::Artifact::App.new(cask).install_phase }
+    }
+
+    let(:source_path) { cask.staged_path.join("Caffeine.app") }
+    let(:target_path) { Hbc.appdir.join("AnotherName.app") }
+
+    before do
       TestHelper.install_without_artifacts(cask)
     end
-  }
 
-  describe "activate to alternate target" do
     it "installs the given apps using the proper target directory" do
-      cask = local_alt_caffeine
+      File.ftype(source_path).must_equal "directory"
```

Comment:
Done in https://github.com/Homebrew/brew/pull/1329/commits/cdd57803a6419bd01416cdbec421973b475f461c.